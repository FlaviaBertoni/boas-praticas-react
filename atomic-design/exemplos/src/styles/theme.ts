const colors = {
  white: '#fff',
  black: '#000',
  primary: '#7E7EB8',
  dark: '#2C2E33',
  disable: '#747A88',
};

const getColorCode = (color?: ThemeColorsKeys) => colors[color || 'primary'];

const isDarkColor = (color?: ThemeColorsKeys) => ![colors.white, colors.disable].includes(color || colors.primary);

const theme = { colors, getColorCode, isDarkColor };

export type ThemeType = typeof theme;
export type ThemeColorsType = typeof colors;
export type ThemeColorsKeys = keyof ThemeColorsType;

export default theme;
