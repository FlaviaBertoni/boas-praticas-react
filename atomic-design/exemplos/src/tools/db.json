{
  "vehicles": [
    {
      "id": "1",
      "model": "K.I.T.T",
      "price": 42,
      "transmission": "Manual",
      "seats": 4,
      "fuel_efficiency": 14.8,
      "efficiency_unit": "km",
      "image": "/images/cars/KITT.png"
    },
    {
      "id": "2",
      "model": "Ecto-1",
      "price": 52.99,
      "transmission": "Manual",
      "seats": 4,
      "fuel_efficiency": 7,
      "efficiency_unit": "km",
      "image": "/images/cars/ecto-1.png"
    },
    {
      "id": "3",
      "model": "DeLorean",
      "price": 52,
      "transmission": "Manual",
      "seats": 2,
      "fuel_efficiency": 12.8,
      "efficiency_unit": "km",
      "image": "/images/cars/delorean.png"
    },
    {
      "id": "4",
      "model": "Bat MÃ³vel",
      "price": 42,
      "transmission": "Auto",
      "seats": 2,
      "fuel_efficiency": 5.8,
      "efficiency_unit": "km",
      "image": "/images/cars/Batmovel.png"
    },
    {
      "id": "5",
      "model": "Herbie",
      "price": 79,
      "transmission": "Manual",
      "seats": 4,
      "fuel_efficiency": 15.6,
      "efficiency_unit": "km",
      "image": "/images/cars/Herbie.png"
    },
    {
      "id": "6",
      "model": "The Mistery Machine",
      "price": 199,
      "transmission": "Auto",
      "seats": 7,
      "fuel_efficiency": 9.5,
      "efficiency_unit": "km",
      "image": "/images/cars/TheMysteryMachine.png"
    }
  ],
  "users": [
    {
      "id": 1,
      "email": "carlostgti@outlook.com",
      "password": "M@n4nc14l",
      "name": "Carlos Queiroz"
    },
    {
      "id": 2,
      "email": "cqueiroz@ciandt.com",
      "password": "12345678",
      "name": "Carlos Queiroz"
    },
    {
      "newUser": {
        "name": "Carlos teste",
        "email": "carlos.da.catia@gmail.com",
        "password": "123456",
        "id": 1615213221105
      },
      "id": 3
    },
    {
      "newUser": {
        "name": "Carlos Queiroz",
        "email": "carlos.da.catia@gmail.com",
        "password": "123456",
        "id": 1615214235093
      },
      "id": 4
    },
    {
      "newUser": {
        "name": "Carlos Queiroz",
        "email": "carlos.da.catia@gmail.com",
        "password": "123456",
        "id": 1615214778055
      },
      "id": 5
    },
    {
      "name": "Carlos Queiroz",
      "email": "carlos.da.catia@gmail.com",
      "password": "123456",
      "id": 1615215407741
    }
  ]
}