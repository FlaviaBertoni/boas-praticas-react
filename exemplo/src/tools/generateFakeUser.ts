const generateFakeUser = () => ({
  name: 'Fake test user',
  email: 'fakeuser@fake.dev',
  password: 'F@k3p4ssw0rd',
});

export default generateFakeUser;
