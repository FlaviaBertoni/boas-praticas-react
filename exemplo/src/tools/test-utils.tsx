import { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from '../styles/theme';

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
