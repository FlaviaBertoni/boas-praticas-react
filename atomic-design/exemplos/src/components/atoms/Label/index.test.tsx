import { render } from "../../../tools/test-utils";
import { Label } from ".";
import theme, { ThemeColorsType } from "../../../styles/theme";

type Size = "small" | "medium" | "large" | "extra-large";
type Weight = "normal" | "bold" | "400" | "700";

const colors = Object.keys(theme.colors);
const sizes = ["small", "medium", "large", "extra-large"];
const weights = ["normal", "bold", "400", "700"];

const matriz: Array<Array<String>> = [];

// Just for testing purpose
colors.map((color) =>
  sizes.map((size) =>
    weights.map((weight) => matriz.push([size, weight, color]))
  )
);

describe("Test case for Label component", () => {
  // We can create tests to check just one property

  it("should be able to render the Label component with small size", () => {
    const { container } = render(<Label size="small">unit test</Label>);

    // these assertions were using jest-style-component lib
    expect(container.firstChild).toHaveStyleRule("font-size", "16px");
    expect(container.firstChild).toHaveStyleRule("color", theme.colors.primary);

    // this assertion was using only jest matcher
    expect(container.firstChild).toHaveStyle({
      fontSize: "16px",
      color: theme.colors.primary,
      fontWeight: "normal",
    });
  });

  // We can test a list of properties using each.
  it.each(colors)(
    "should be able to render the Label component for %s color and match with the snapshot",
    (color) => {
      const { container } = render(
        <Label size="small" color={color as keyof ThemeColorsType}>
          unit test
        </Label>
      );

      expect(container).toMatchSnapshot();
    }
  );

  // We can even test a list of lists using each:
  it.each(matriz)(
    "should be able to render the Label component with %s size, %s weight, %s color and match with the snapshot",
    (size, color, weight) => {
      const { container } = render(
        <Label
          size={size as Size}
          color={color as keyof ThemeColorsType}
          weight={weight as Weight}
        >
          unit test
        </Label>
      );

      expect(container).toMatchSnapshot();
    }
  );

  // Also we can use a inline snapshot
  it("should be able to render", () => {
    const { container } = render(<Label size="extra-large">unit test</Label>);

    expect(container).toMatchInlineSnapshot(`
      .c0 {
        font-size: 48px;
        color: #7E7EB8;
        font-weight: normal;
      }

      <div>
        <label
          class="c0"
        >
          unit test
        </label>
      </div>
    `);
  });
});
