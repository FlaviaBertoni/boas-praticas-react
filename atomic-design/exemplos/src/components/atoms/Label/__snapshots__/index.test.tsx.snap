// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for Label component should be able to render the Label component for black color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  color: #000;
  font-weight: normal;
}

<div>
  <label
    class="c0"
    color="black"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component for dark color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  color: #2C2E33;
  font-weight: normal;
}

<div>
  <label
    class="c0"
    color="dark"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component for disable color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  color: #747A88;
  font-weight: normal;
}

<div>
  <label
    class="c0"
    color="disable"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component for primary color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  color: #7E7EB8;
  font-weight: normal;
}

<div>
  <label
    class="c0"
    color="primary"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component for white color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  color: #fff;
  font-weight: normal;
}

<div>
  <label
    class="c0"
    color="white"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with extra-large size, bold weight, black color and match with the snapshot 1`] = `
.c0 {
  font-size: 48px;
  font-weight: black;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with extra-large size, bold weight, dark color and match with the snapshot 1`] = `
.c0 {
  font-size: 48px;
  font-weight: dark;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with extra-large size, bold weight, disable color and match with the snapshot 1`] = `
.c0 {
  font-size: 48px;
  font-weight: disable;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with extra-large size, bold weight, primary color and match with the snapshot 1`] = `
.c0 {
  font-size: 48px;
  font-weight: primary;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with extra-large size, bold weight, white color and match with the snapshot 1`] = `
.c0 {
  font-size: 48px;
  font-weight: white;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with extra-large size, normal weight, black color and match with the snapshot 1`] = `
.c0 {
  font-size: 48px;
  font-weight: black;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with extra-large size, normal weight, dark color and match with the snapshot 1`] = `
.c0 {
  font-size: 48px;
  font-weight: dark;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with extra-large size, normal weight, disable color and match with the snapshot 1`] = `
.c0 {
  font-size: 48px;
  font-weight: disable;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with extra-large size, normal weight, primary color and match with the snapshot 1`] = `
.c0 {
  font-size: 48px;
  font-weight: primary;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with extra-large size, normal weight, white color and match with the snapshot 1`] = `
.c0 {
  font-size: 48px;
  font-weight: white;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with large size, bold weight, black color and match with the snapshot 1`] = `
.c0 {
  font-size: 36px;
  font-weight: black;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with large size, bold weight, dark color and match with the snapshot 1`] = `
.c0 {
  font-size: 36px;
  font-weight: dark;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with large size, bold weight, disable color and match with the snapshot 1`] = `
.c0 {
  font-size: 36px;
  font-weight: disable;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with large size, bold weight, primary color and match with the snapshot 1`] = `
.c0 {
  font-size: 36px;
  font-weight: primary;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with large size, bold weight, white color and match with the snapshot 1`] = `
.c0 {
  font-size: 36px;
  font-weight: white;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with large size, normal weight, black color and match with the snapshot 1`] = `
.c0 {
  font-size: 36px;
  font-weight: black;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with large size, normal weight, dark color and match with the snapshot 1`] = `
.c0 {
  font-size: 36px;
  font-weight: dark;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with large size, normal weight, disable color and match with the snapshot 1`] = `
.c0 {
  font-size: 36px;
  font-weight: disable;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with large size, normal weight, primary color and match with the snapshot 1`] = `
.c0 {
  font-size: 36px;
  font-weight: primary;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with large size, normal weight, white color and match with the snapshot 1`] = `
.c0 {
  font-size: 36px;
  font-weight: white;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with medium size, bold weight, black color and match with the snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-weight: black;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with medium size, bold weight, dark color and match with the snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-weight: dark;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with medium size, bold weight, disable color and match with the snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-weight: disable;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with medium size, bold weight, primary color and match with the snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-weight: primary;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with medium size, bold weight, white color and match with the snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-weight: white;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with medium size, normal weight, black color and match with the snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-weight: black;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with medium size, normal weight, dark color and match with the snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-weight: dark;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with medium size, normal weight, disable color and match with the snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-weight: disable;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with medium size, normal weight, primary color and match with the snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-weight: primary;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with medium size, normal weight, white color and match with the snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-weight: white;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with small size, bold weight, black color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  font-weight: black;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with small size, bold weight, dark color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  font-weight: dark;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with small size, bold weight, disable color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  font-weight: disable;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with small size, bold weight, primary color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  font-weight: primary;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with small size, bold weight, white color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  font-weight: white;
}

<div>
  <label
    class="c0"
    color="bold"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with small size, normal weight, black color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  font-weight: black;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with small size, normal weight, dark color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  font-weight: dark;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with small size, normal weight, disable color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  font-weight: disable;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with small size, normal weight, primary color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  font-weight: primary;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;

exports[`Test case for Label component should be able to render the Label component with small size, normal weight, white color and match with the snapshot 1`] = `
.c0 {
  font-size: 16px;
  font-weight: white;
}

<div>
  <label
    class="c0"
    color="normal"
  >
    unit test
  </label>
</div>
`;
